src/robusta/core/reporting/blocks.py

```diff
@@ -492,12 +494,19 @@ def __init__(self, data: PrometheusQueryResult, query: str,
         super().__init__(data=data, metadata=metadata, vertical_lines=vertical_lines, horizontal_lines=horizontal_lines,
                          y_axis_type=y_axis_type, graph_name=graph_name)
 
+        self.set_metrics_legends_labels(legends_labels=metrics_legends_labels)
```

Comment:
I think we don't need a class method, and don't need to add that key if the legends are None
I would just do here:
```
if metrics_legends_labels:
  self.metadata["metrics_legends_labels"] =  json.dumps(metrics_legends_labels)
```